### PR TITLE
perf: increase performance of nx console

### PR DESCRIPTION
**Pull request by Cammisuli**
This removes JSON5 parsing to use regular `JSON.parse`. There might be some json files that allow comments, so we have `stripJsonComments` included as well. 

This is the same behaviour that's in core Nx, so if there is any configuration file that has JSON5 specs in it, Nx will fail anyway ☹️ 

Here's a gif with these changes on a workspace that has 1426 projects with the following specs:
image

nx-console-perf-fix 2021-03-12 13_50_42

### CODE REVIEW

File: libs/server/src/lib/utils/utils.ts [hunk 1]
```diff
@@ -131,7 +131,7 @@ export function fileExistsSync(filePath: string): boolean {
 }
 
 export function readAndParseJson(fullFilePath: string): any {
-  return JSON5.parse(readFileSync(fullFilePath).toString());
+  return JSON.parse(stripJsonComments(readFileSync(fullFilePath).toString()));
```
Comment (sandikbarr) [inline libs/server/src/lib/utils/utils.ts:134]: mentioned this to Isaac and he wondered if it should also try catch and fallback to JSON5 if the regular parse fails

Comment (Cammisuli) [reply libs/server/src/lib/utils/utils.ts:134]: This is the same "technique" used in Nx. If the workspace.json file includes anything related to json5 (trailing commas, etc) then Nx itself won't work anyway ☹️ 

I still want to work on this though, I think making this asynchronous will help with other things.

**Comment by isaacplmann** [comment]
It could be that this is present to handle plain angular CLI configs as well.

File: apps/vscode/src/package.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
     "Other"
   ],
   "activationEvents": [
-    "*"
+    "workspaceContains:nx.json"
```
Comment (Cammisuli) [inline apps/vscode/src/package.json:40]: I also made this only activate when you have a nx.json file in your vscode workspace. This should free up memory for folks who use vscode for things that aren't Nx 👀

**Review by sandikbarr — APPROVED** [review]
🎉

**Pull request merged**